### PR TITLE
V3.0.15 carto memfixes

### PR DESCRIPTION
**Pull request by Algunenano**
Done

### CODE REVIEW

File: include/mapnik/svg/svg_path_attributes.hpp [hunk 1]
```diff
@@ -34,6 +34,8 @@
 #include "agg_trans_affine.h"
 #pragma GCC diagnostic pop
 
+#include <tuple>
```
Comment (rafatower) [inline include/mapnik/svg/svg_path_attributes.hpp:37]: minor thing: isn't this an implementation detail to be moved to the cpp file?

**Comment by Algunenano** [comment]
This should fix the memory issues and add stability to the tests.

I've updated the images: [Here is a comparison vs no cache](https://github.com/CartoDB/test-data-visual/compare/2f22a10fbaa19dd12e078b39e3c2a9f579ea8c2c...3.0.x-carto). I've made several runs with my main machine and with a 12.04 build and both generate the same images all the time.

PENDING:
- Confirm that the tests are stable in Travis.

File: src/agg/process_markers_symbolizer.cpp [hunk 2]
```diff
@@ -128,16 +133,15 @@ struct agg_markers_renderer_context : markers_renderer_context
 
                     // Build local transformation matrix by resetting translation coordinates
                     agg::trans_affine marker_tr_copy(marker_tr);
-                    marker_tr_copy.tx = sample_x / sampling_rate - x0;
-                    marker_tr_copy.ty = sample_y / sampling_rate - y0;
+                    marker_tr_copy.tx = dx - x0;
+                    marker_tr_copy.ty = dy - y0;
 
                     // Create fill image
-                    std::shared_ptr<image_rgba8> fill_img;
```
Comment (rafatower) [inline src/agg/process_markers_symbolizer.cpp:135]: I think redefining this as `rgba8` was part of the optimization.

**Review by rafatower — COMMENTED** [review]
Just a few comments and nothing blocking, except the tests themselves (which I think you fixed while I was taking a look at this)

File: include/mapnik/svg/svg_path_attributes.hpp (hunk 1 above)
Comment (Algunenano) [reply include/mapnik/svg/svg_path_attributes.hpp:37]: There isn't a .cpp file, everything is defined in the header file.

File: src/agg/process_markers_symbolizer.cpp (hunk 2 above)
Comment (Algunenano) [reply src/agg/process_markers_symbolizer.cpp:135]: I didn't change the definition, just changed places (both are and were `std::shared_ptr<image_rgba8>`).

**Comment by Algunenano** [comment]
Closing in favor of https://github.com/CartoDB/mapnik/pull/36. Let me know if you have any comments or questions.

**Pull request closed without merging**